### PR TITLE
feat(behavior_path_planner): keep original points in resampling of pull over

### PR DESCRIPTION
**Pull request by kosuke55**
Signed-off-by: kosuke55 <kosuke.tnp@gmail.com>

## Description

- keep original input poins in resampling of end of behavior_path_planner.
- add flag for keeping oringal points to `resamplePathWithSpline`
  - set `false` defaultly, so in each scene module do not keep the orintal points unintentionally (not to increase calculation cost)

## Related links

[tier4 internal slack](https://star4.slack.com/archives/C03QW0GU6P7/p1670491839630099)

## Tests performed

psim, real vehicle

[tier4 internal scenario test](https://evaluation.tier4.jp/evaluation/reports/e9b78c23-c2cc-57d6-8d0e-ba55c81b28de?project_id=prd_jt)
821/825 -> 821/825

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: planning/behavior_path_planner/src/path_utilities.cpp [hunk 1]
```diff
@@ -66,7 +67,7 @@ PathWithLaneId resamplePathWithSpline(const PathWithLaneId & path, double interv
     transformed_path.at(i) = path.points.at(i).point;
   }
 
-  constexpr double epsilon = 0.01;
+  constexpr double epsilon = 0.1;
```
Comment (kosuke55) [inline planning/behavior_path_planner/src/path_utilities.cpp:70]: 0.01 seems too close and sometimes causes path distortion.

Comment (kosuke55) [reply planning/behavior_path_planner/src/path_utilities.cpp:70]:
```suggestion
  constexpr double epsilon = 0.2;
```

is better from @TakaHoribe

**Comment by kosuke55** [comment]
[tier4 internal scenario test](https://evaluation.tier4.jp/evaluation/reports/e9b78c23-c2cc-57d6-8d0e-ba55c81b28de?project_id=prd_jt)
821/825 -> 821/825

**Comment by kosuke55** [comment]
@TakaHoribe 
Could you please approve this?

**Review by TakaHoribe — APPROVED** [review]
This is a temporal solution. Will be redesigned in the future.

**Pull request merged**